src/utils/script.ts

```diff
@@ -1,15 +1,30 @@
 import { CharacterNameSchema, characterRegistry } from "@/data/characters/registry";
 import type { Script } from "@/types/script";
+import type { HomebrewCharacter } from "@/data/characters/homebrew";
 import type { CollectionEntry } from "astro:content";
 import { match } from "ts-pattern";
 
 export const parseScriptData = (script: CollectionEntry<"scripts">): Script => {
   const [meta, ...characters] = script.data;
 
-  const characterBreakdown = characters.reduce(
-    (acc, character) => {
-      const parsedCharacter = CharacterNameSchema.parse(character);
+  return {
+    id: script.id,
+    name: meta.name,
+    author: meta.author,
+    characterBreakdown: meta.almanac
+      ? homebrewCharacterBreakdown(characters as HomebrewCharacter[])
+      : standardCharacterBreakdown(characters as string[]),
+    characters: meta.almanac
+      ? (characters as HomebrewCharacter[])
+      : characters.map((character) => CharacterNameSchema.parse(character as string)),
```

Comment:
not for this PR, but we can definitely iterate on this. If almanac is the source of truth for what is homebrew and what isn't, we should be able to update the zod schema to just be a big union of either format. I don't think there's anything to change here right now the type casts are fine, just food for thought for the future

Especially if @MeJosh might be working on more schema updates, things in here could get messy quick and we could lose type-safety